tests/unit-tests/13-modules/#.scm

```diff
@@ -17,13 +17,39 @@
                              calls))))
            namespaces)))
 
+(define ##output-string-port (open-string))
+
+(define (call-with-output-port port thunk)
+  (include "~~lib/_repl#.scm")
+  (implement-type-repl-channel)
+  (let* ((repl-channel (##thread-repl-channel-get! (current-thread)))
+         (old-output-port (macro-repl-channel-output-port repl-channel)))
+    (macro-repl-channel-output-port-set! repl-channel port)
+    (parameterize
+      ((current-output-port port))
+      (let ((result (thunk port)))
+        (macro-repl-channel-output-port-set! repl-channel old-output-port)
+        result))))
+
+
 (define (capture-behavior thunk)
-  (with-exception-catcher
-   (lambda (e) e)
-   thunk))
+  (call-with-output-port
+    ##output-string-port
+    (lambda (port)
+      (with-exception-catcher
+        (lambda (e) e)
+        (lambda ()
+          (let ((result (thunk)))
+            (cons result (get-output-string port))))))))
 
 (define (compare-behavior namespaces calls . behaviors)
 
+  (define (behavior->string obj)
+    (let ((str (object->string obj)))
```

Comment:
```
> (object->string (iota 20))
"(0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19)"
> (object->string (iota 20) 30)
"(0 1 2 3 4 5 6 7 8 9 10 11 ..."
```

Reply:
I forgot about the second parameter of `object->string` which does exactly what I wanted.